onnxruntime/contrib_ops/cpu/bert/attention.cc

```diff
@@ -238,8 +238,13 @@ Status Attention<T>::Compute(OpKernelContext* context) const {
       float* x = reinterpret_cast<T*>(scratch_data) + j * D;
       float* y = x;
 
-      for (int i = 0; i < D; i++)
-        y[i] = expf(x[i]);
+      float max = 0.f;
+      for (int i = 0; i < D; i++) {
+        if (max < x[i]) max = x[i];
+      }
+      for (int i = 0; i < D; i++) {
+        y[i] = expf(x[i] - max);
+      }
```

Comment:
Would it be more performant to use Eigen for reduce max and computation?

Reply:
Softmax is very cheap, it's hard to see any difference.

Reply:
Actually, it is slower with Eigen.